### PR TITLE
changing the selected variable option when profile or permission type is changed

### PR DESCRIPTION
**Pull request by emre2038**
## Description
changing the selected variable when profile or permission type is changed while adding/modifying profile indicators

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-325

## How to test it locally
* go to add indicator page
* select a variable
* change the selected profile / permission type
* confirm that the selected variable is `---------`

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [ ]  📖 changelog filled out

### Commits

- [x]  commits are clean
- [x]  commit messages are clean

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers
- [ ]  black was run locally (as part of the pre-commit hook)

### Testing

- [ ]  ✅ added (appropriate) unit tests
- [x]  💢 edge cases in tests were considered
- [x]  ✅ ran tests locally & are passing

### CODE REVIEW

**Review by michaelglenister — APPROVED** [review]
Thanks, fixes the problem and ready to merge

**Review by goyal1092 — CHANGES_REQUESTED** [review]
Hey Emre
I was testing on this local and found out an issue in re-opening saved profile indicator
Screen Shot 2022-03-28 at 12 12 35 PM
r

**Comment by goyal1092** [comment]
can you please also add a test if accurate variable and permission radio is selected?
I would advise writing a unit test for this. Usually, we check for the initial value set in the Django form field in this case.
But as this is a custom widget so we will have to test `VariableFilterWidget`.
There should be a way to test using django forms though.
Example: https://github.com/OpenUpSA/wazimap-ng/blob/staging/tests/datasets/admin/test_dataset_admin.py#L172

I would also recommend adding a test for queryset that we are sending and add a test specifically for ProfileAdmin to make sure we show only required data to admin user

**Comment by goyal1092** [comment]
Issue in Profile Key Metric Page
As this widget is being used in Profile Key metrics and Profile Highlights we have changed js for those as well
There is something weird happening with key metrics. 
Options are not getting disabled when profile or permissions are changed
Screen Shot 2022-03-28 at 12 30 33 PM
Screen Shot 2022-03-28 at 12 30 33 PM

**Comment by emre2038** [comment]
bugs are fixed 
widget is working in profile highlight & profile key metrics pages too

File: wazimap_ng/general/widgets.py [hunk 1]
```diff
@@ -63,10 +68,15 @@ def get_context(self, name, value, attrs=None):
             value = queryset.get(id=value)
             selected_permission = value.dataset.permission_type
 
+        profile_id = None
+        if self.instance is not None:
+            profile_id = self.instance.profile_id
+
```
Comment (goyal1092) [inline wazimap_ng/general/widgets.py:74]: I think we can filter queryset according to profile_id
Q(indicator__dataset__permission_type="public") | Q(indicator__dataset__permision_type="private", dataset_profile_id=profile_id)

File: wazimap_ng/general/templates/widgets/VariableFilterWidget.html [hunk 2]
```diff
@@ -1,7 +1,12 @@
 <div id="variable-permission-filter">
     {{ choice_field }}
 </div>
-<select id="id_indicator" required name="{{ name }}">
+<input type="hidden" name="selected_profile_id" value="{{ profile_id }}">
```
Comment (goyal1092) [inline wazimap_ng/general/templates/widgets/VariableFilterWidget.html:4]: We could add a hidden field using Django forms if required

**Review by goyal1092 — CHANGES_REQUESTED** [review]

File: wazimap_ng/general/templates/widgets/VariableFilterWidget.html (hunk 2 above)
Comment (emre2038) [reply wazimap_ng/general/templates/widgets/VariableFilterWidget.html]: removed this field now that we do not need profile id in frontend

File: wazimap_ng/general/widgets.py (hunk 1 above)
Comment (emre2038) [reply wazimap_ng/general/widgets.py:78]: done 👍

**Comment by emre2038** [comment]
I liked removing filtering by profile Id from `widgets.py` and turning it into a method @goyal1092

File: tests/profile/admin/test_profile_indicator_admin.py [hunk 3]
```diff
@@ -47,6 +96,117 @@ def test_indicator_queryset(self, mocked_request, indicator, qualitative_indicat
         # assert count
```
Comment (michaelglenister) [inline tests/profile/admin/test_profile_indicator_admin.py:96]: Not related to this change but this seems like a planning comment that was forgotten to be removed

File: tests/profile/admin/test_profile_indicator_admin.py [hunk 4]
```diff
@@ -47,6 +96,117 @@ def test_indicator_queryset(self, mocked_request, indicator, qualitative_indicat
...
+
+        assert res.status_code == 200
+        form = res.context["adminform"].form
+        errors = form.errors
+        assert (
+            errors["indicator"].data[0].message
+            == "Select a valid choice. That choice is not one of the available choices."
+        )
+
+    def test_adding_private_indicator_from_different_profile(
```
Comment (michaelglenister) [inline tests/profile/admin/test_profile_indicator_admin.py:192]: I think you have covered all cases where a private variable should not be accessible, what about a test case where it should be accessible?

**Review by michaelglenister — COMMENTED** [review]
I'm happy with the changes. There is one comment about a possible additional test case for when private variables should be accessible if you agree that it's helpful. Either way doesn't need another code review, you can move to PO review. Thanks

File: tests/profile/admin/test_profile_indicator_admin.py (hunk 4 above)
Comment (emre2038) [reply tests/profile/admin/test_profile_indicator_admin.py:192]: added 2 more tests to cover accesibility of the indicators 👍

**Pull request merged**